### PR TITLE
refactor: Cluster features for GNN pipeline

### PR DESCRIPTION
**Pull request by benjaminhuth**
This refactors the way (cluster)features for the GNN pipeline are computed. Also extends the `Examples::Cluster` to support a global position which is needed by athena data (Updates to that will be merged later).

### CODE REVIEW

File: Examples/Algorithms/TrackFindingExaTrkX/src/TrackFindingAlgorithmExaTrkX.cpp [hunk 1]
```diff
@@ -119,6 +119,14 @@ class ExamplesEdmHook : public Acts::ExaTrkXHook {
   }
 };
 
+// TODO do we have these function in the repo somewhere?
+float theta(float r, float z) {
+  return std::atan2(r, z);
+}
+float eta(float r, float z) {
+  return -std::log(std::tan(0.5 * theta(r, z)));
+}
```
Comment (andiwand) [inline Examples/Algorithms/TrackFindingExaTrkX/src/TrackFindingAlgorithmExaTrkX.cpp:128]: can this be helpful https://github.com/acts-project/acts/blob/abebe1f42c022f5ecfa0db1b434f8451be25c45e/Core/include/Acts/Utilities/VectorHelpers.hpp#L85-L127

I fear the `-std::log(std::tan(0.5 * theta(r, z)))` is hardcoded a couple of times in the repo. We could clean this out and replace this with a theta input once

File: Examples/Algorithms/TrackFindingExaTrkX/src/TrackFindingAlgorithmExaTrkX.cpp [hunk 2]
```diff
@@ -200,54 +219,70 @@ ActsExamples::ProcessCode ActsExamples::TrackFindingAlgorithmExaTrkX::execute(
...
     // I would prefer to use a std::span or boost::span here once available
-    float* featurePtr = features.data() + i * numFeatures;
+    float* f = features.data() + isp * numFeatures;
+
+    using NF = NodeFeature;
+
+    for (auto ift = 0ul; ift < numFeatures; ++ift) {
+      // clang-format off
+      switch(m_cfg.nodeFeatures[ift]) {
+        break; case NF::eR:           f[ift] = std::hypot(sp.x(), sp.y());
```
Comment (andiwand) [inline Examples/Algorithms/TrackFindingExaTrkX/src/TrackFindingAlgorithmExaTrkX.cpp:261]: wow that is new to me: `break` before `case`

**Review by andiwand — APPROVED** [review]
LGTM - left a few minor comments

File: Examples/Algorithms/TrackFindingExaTrkX/src/TrackFindingAlgorithmExaTrkX.cpp (hunk 1 above)
Comment (benjaminhuth) [reply Examples/Algorithms/TrackFindingExaTrkX/src/TrackFindingAlgorithmExaTrkX.cpp:128]: I try to make this more consistent in a subsequent PR

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [b5069e1d0130f8b4208d81785c84459153ddf4d2]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/41074635) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7732034)!

**Pull request merged**